snapcraft/storeapi/__init__.py

```diff
@@ -643,11 +640,20 @@ def push_assertion(self, snap_id, assertion, endpoint):
                 'snap_developer': assertion.decode('utf-8'),
             }
         auth = _macaroon_auth(self.conf)
+
+        url = 'snaps/{}/{}'.format(snap_id, endpoint)
+
+        # For `snap-developer`, revoking developers will require their uploads
+        # to be invalidated.
+        if force:
+            url = url + '?ignore_revoked_uploads'
```

Comment:
hmm, this I did not double check, should probably be passed on using a dict for on `self.put`